src/stackwalk.c

```diff
@@ -218,12 +218,22 @@ NOINLINE size_t rec_backtrace(jl_bt_element_t *bt_data, size_t maxsize, int skip
     bt_cursor_t cursor;
     if (!jl_unw_init(&cursor, &context) || maxsize == 0)
         return 0;
-    jl_gcframe_t *pgcstack = jl_pgcstack;
+    jl_gcframe_t *pgcstack = no_gcstack ? NULL : jl_pgcstack;
     size_t bt_size = 0;
     jl_unw_stepn(&cursor, bt_data, &bt_size, NULL, maxsize, skip + 1, &pgcstack, 0);
     return bt_size;
 }
 
+NOINLINE int failed_to_sample_task_fun(jl_bt_element_t *bt_data, size_t maxsize, int skip) JL_NOTSAFEPOINT
+{
+    return rec_backtrace(bt_data, maxsize, skip, 1);
```

Comment:
Is there a reason you don't just use the pointer to it as a sort of sentinel value directly instead of having `jl_unw_get` to do it for you:
```suggestion
    if (maxsize < 1) return 0;
    bt_data[0].uintptr = (uintptr_t) &failed_to_sample_task_fun;
    return 1;
```

We could also make actual sentinel values, since we know that the zero page cannot be mapped that many of those values are legal (whereas Profile.flatten starts from -1 and counts down to make fake frames)

Reply:
> Is there a reason you don't just use the pointer to it as a sort of sentinel value directly instead of having jl_unw_get to do it for you

Not really.

What you're proposing should work as well. Will try it.

Reply:
Applied in the latest commit.